github_connector_odoo/models/github_repository_branch.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016-Today: Odoo Community Association (OCA)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+import os
+
+from os.path import join as opj
+
+from openerp import models, fields, api
+
+from openerp.modules import load_information_from_description_file
+from openerp.modules.module import MANIFEST
+
+_logger = logging.getLogger(__name__)
+
+
+class GithubRepositoryBranch(models.Model):
+    _inherit = ['github.repository.branch']
+
+    module_paths = fields.Text(
+        string='Module Paths', help="Set here extra relative paths"
+        " you want to scan to find modules. If not set, root path will be"
+        " scanned. One repository per line. Exemple:\n"
+        "./addons/\n"
+        "./openerp/addons/")
+
+    module_version_ids = fields.One2many(
+        comodel_name='odoo.module.version',
+        inverse_name='repository_branch_id', string='Module Versions')
+
+    module_version_qty = fields.Integer(
+        string='Module Versions Quantity',
+        compute='_compute_module_version_qty')
+
+    coverage_url = fields.Char(
+        string='Coverage Url', store=True, multi='complete_name',
+        compute='_compute_multi_from_complete_name')
+
+    coverage_image_url = fields.Char(
+        string='Coverage Image Url', store=True, multi='complete_name',
+        compute='_compute_multi_from_complete_name')
+
+    integration_service_url = fields.Char(
+        string='Integration Service Url', store=True, multi='complete_name',
+        compute='_compute_multi_from_complete_name')
+
+    integration_service_image_url = fields.Char(
+        string='Integration Service Image Url', store=True,
+        multi='complete_name',
+        compute='_compute_multi_from_complete_name')
+
+    runbot_url = fields.Char(
+        string='Runbot Url', multi='complete_name',
+        compute='_compute_multi_from_complete_name')
+
+    runbot_image_url = fields.Char(
+        string='Runbot Image Url',
+        multi='complete_name',
+        compute='_compute_multi_from_complete_name')
+
+    github_url = fields.Char(
+        string='Github Url', store=True,
+        multi='complete_name',
+        compute='_compute_multi_from_complete_name')
+
+    # Compute Section
+    @api.multi
+    @api.depends('name', 'repository_id.complete_name', 'repository_id.ci_id')
+    def _compute_multi_from_complete_name(self):
+        for branch in self:
+            # Compute Coverage Service Url and Image Url
+            branch.coverage_url =\
+                'https://coveralls.io/github/' +\
```

Comment:
Might be best to switch this to Codecov, we're switching from Coveralls at some point because it's not as good with the changed check